### PR TITLE
Created a basic admin page

### PR DESCRIPTION
**Pull request by samiwel**
### What is the context of this PR?
Introduced basic admin page.
Added necessary GraphQL Apollo container logic to retrieve and delete the relevant Questionnaire data.

image

### How to review 
All tests should pass.
This should not be merged yet, but have raised PR so that team can review and have visibility of changes.

### CODE REVIEW

**Comment by samiwel** [comment]
[Questionnaires Admin (Basic) (5)](https://trello.com/c/koxQ1wY7/114-questionnaires-admin-basic-5)

File: src/containers/QuestionnairesPage/QuestionnairesPage.js [hunk 1]
```diff
@@ -1,35 +1,311 @@
...
+                      />}
+                  </TD>
+                </TR>
+              </CSSTransition>
+            );
+          })}
+        </TransitionGroup>
+      </Table>
+    );
+  } else {
```
Comment (WickyNilliams) [inline src/containers/QuestionnairesPage/QuestionnairesPage.js]: Can you invert the logic of this if statement? That way we have nesting and it's easier to follow (imo)

```js
if (!questionnaires || questionnaires.length === 0) {
     return <p>You have no questionnaires</p>;
} 

// carry on rendering as normal
```

File: src/containers/QuestionnairesPage/QuestionnairesPage.js [hunk 2]
```diff
@@ -1,35 +1,311 @@
...
+          <ExistingQuetionnairesTable {...props} />
+        </CenteredPanel>
+      </Center>
+    </BaseLayout>
+  );
+};
 
+Questionnaires.propTypes = {
+  questionnaires: questionnairesListPropType
+};
```
Comment (WickyNilliams) [inline src/containers/QuestionnairesPage/QuestionnairesPage.js]: Why is everything bundled into a single file? You could e.g. extract the Table component and all it's bits and pieces into a regular component

File: src/containers/QuestionnairesPage/QuestionnairesPage.test.js [hunk 3]
```diff
@@ -6,7 +6,13 @@ let wrapper;
 
 describe("containers/Questionnaires", () => {
   beforeEach(() => {
-    wrapper = mountWithRouter(<Questionnaires />);
+    const questionnaires = [];
+    wrapper = mountWithRouter(
```
Comment (WickyNilliams) [inline src/containers/QuestionnairesPage/QuestionnairesPage.test.js]: Can we actually get away with shallow rendering here?

File: src/containers/QuestionnairesPage/index.js [hunk 4]
```diff
@@ -1,7 +1,11 @@
...
 import Questionnaires from "./QuestionnairesPage";
+import withQuestionnaireList from "../enhancers/withQuestionnaireList";
+import withDeleteQuestionnaire from "../enhancers/withDeleteQuestionnaire";
 
-const mapStateToProps = (state, ownProps) => ({});
-const mapDispatchToProps = {};
-
-export default connect(mapStateToProps, mapDispatchToProps)(Questionnaires);
+export default compose(
+  withRouter,
```
Comment (WickyNilliams) [inline src/containers/QuestionnairesPage/index.js]: Do we need the router? We don't seem to use `history` or anything like that. Or is it needed for something else?

File: src/containers/enhancers/withQuestionnaireList.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+  const questionnaires = data.questionnaires.map(questionnaire => {
+    return merge({}, questionnaire, {
+      // TODO Replace this when we have mechanism for retrieving comments.
+      status: "Unpublished",
+      comments: {
+        unread: true,
+        count: 1
+      },
+      actions: {
```
Comment (WickyNilliams) [inline src/containers/enhancers/withQuestionnaireList.js]: What's this actions obj about?

File: src/containers/QuestionnairesPage/QuestionnairesPage.js [hunk 6]
```diff
@@ -1,35 +1,311 @@
...
+    top: -2px;
+    right: -4px;
+  `;
+
+  return (
+    <StyledSpan>
+      <StyledButton title={"Click to see comments"} onClick={handleClick} />
+      {hasUnread &&
+        <UnreadCommentsNotifier title={"You have unread comments"} />}
+    </StyledSpan>
```
[the remark this answers is not in the document]
Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesPage.js]: I did try to use the tooltip but the styling got really messed up, will revisit and have another look at it tomorrow.

File: src/containers/QuestionnairesPage/QuestionnairesPage.js [hunk 7]
```diff
@@ -1,35 +1,311 @@
...
+  );
+};
```
[the remark this answers is not in the document]
Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesPage.js]: Could hide it, depends if they want to use it for user testing or not.

File: src/containers/QuestionnairesPage/QuestionnairesPage.js [hunk 8]
```diff
@@ -1,35 +1,311 @@
...
+        </thead>
+
+        <TransitionGroup enter={false} component={TBody}>
+          {questionnaires.map(questionnaire => {
+            const date = new Date(questionnaire.createdAt);
+            const dd = date.getDate();
+            const mm = date.getMonth() + 1;
+            const yyyy = date.getFullYear();
+
+            const url = `/questionnaire/${questionnaire.id}/design/${questionnaire
```
[the remark this answers is not in the document]
Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesPage.js]: Yeah, it needs an API change similar to http://graphql.org/learn/pagination/ but it was just overkill to make that change as part of this PR.

File: src/containers/enhancers/withQuestionnaireList.js (hunk 5 above)
Comment (samiwel) [reply src/containers/enhancers/withQuestionnaireList.js]: General purpose. Right now it just controls if the delete button is displayed.

File: src/containers/QuestionnairesPage/index.js (hunk 4 above)
Comment (samiwel) [reply src/containers/QuestionnairesPage/index.js]: I'm not sure if it needs the router. The component uses `NavLink` from the `react-router-dom` module. I will remove and see if it breaks anything! 👍

File: scripts/clean-build-all.sh [hunk 9]
```diff
@@ -0,0 +1,16 @@
+#/bin/bash
+
```
[the remark this answers is not in the document]
Comment (samiwel) [reply scripts/clean-build-all.sh]: Yeah, probably. That just caught me out again!

File: src/containers/QuestionnairesPage/QuestionnairesPage.js [hunk 10]
```diff
@@ -1,35 +1,311 @@
...
+    text-align: left;
+  }
+`;
+
+const CommentsButton = ({ hasUnread }) => {
+  const handleClick = () => {
+    alert("Not implemented yet.");
+  };
+
+  const StyledSpan = styled.span`
```
[the remark this answers is not in the document]
Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesPage.js]: Good spot, will move these.

File: src/containers/QuestionnairesPage/QuestionnairesPage.js [hunk 11]
```diff
@@ -1,35 +1,311 @@
...
+  <tbody>
+    {children}
+  </tbody>;
+
+TBody.propTypes = {
+  children: PropTypes.arrayOf(PropTypes.element).isRequired
+};
+
+const ExistingQuetionnairesTable = props => {
+  const handleDelete = questionnaireId => {
```
[the remark this answers is not in the document]
Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesPage.js]: `questionnaireId` is not available on the props, so we need to pass it in here. The props contain a list of questionnaires, so it wouldn't be possible to determine which questionnaire we're deleting without passing in the Id.

File: src/containers/QuestionnairesPage/QuestionnairesPage.js [hunk 12]
```diff
@@ -1,35 +1,311 @@
...
+                    <Link
+                      to={url}
+                      title={questionnaire.title}
+                      aria-label={questionnaire.title}
+                    >
+                      {questionnaire.title}
+                    </Link>
+                  </TD>
+                  <TD>
+                    {`${dd}/${mm}/${yyyy}`}
```
[the remark this answers is not in the document]
Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesPage.js]: Thanks, this is the function I was looking for on the Date API but didn't know what it was called!

File: scripts/clean-build-all.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (samiwel) [reply scripts/clean-build-all.sh]: Have added to the script.

File: src/containers/QuestionnairesPage/index.js (hunk 4 above)
Comment (samiwel) [reply src/containers/QuestionnairesPage/index.js]: Turns out the `withRouter` wasn't needed, so I removed it.

File: src/containers/QuestionnairesPage/QuestionnairesTable.js [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+import { TransitionGroup, CSSTransition } from "react-transition-group";
+import { NavLink } from "react-router-dom";
+
+import { colors } from "../../constants/theme";
+import CommentsButton from "./CommentsButton";
+import DeleteQuestionnaireButton from "./DeleteQuestionnaireButton";
+
+const duration = 300;
+
+const Link = styled(NavLink)`
```
Comment (WickyNilliams) [inline src/containers/QuestionnairesPage/QuestionnairesTable.js]: Why use NavLink vs regular anchor? You'll never get activeClass stuff on this listing page, so seems unnecessary

File: src/containers/QuestionnairesPage/QuestionnairesTable.js [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+`;
+
+const TR = styled.tr`
+  border-top: 1px solid #e2e2e2;
+  opacity: 1;
+  height: 2em;
+
+  &.row-exit {
+    opacity: 0.01;
+    height: 0em;
```
Comment (WickyNilliams) [inline src/containers/QuestionnairesPage/QuestionnairesTable.js]: Height doesn't seem to transition when animating

File: src/containers/QuestionnairesPage/QuestionnairesTable.js [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+  display: inline-block;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  &:hover {
+    opacity: .7;
+  }
+`;
+
+const Table = styled.table`
```
Comment (WickyNilliams) [inline src/containers/QuestionnairesPage/QuestionnairesTable.js]: I wonder whether this should also be extracted into a generic Table component, and then stuff specific to listing Questionnaires overridden here.

File: src/containers/QuestionnairesPage/CommentsButton.js [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+  border: 0;
+  opacity: .7;
+  
+  &:hover { 
+    opacity: 1;
+    transition: opacity 150ms ease-in;
+  }
+`;
+
+const UnreadCommentsNotifier = styled.span`
```
Comment (WickyNilliams) [inline src/containers/QuestionnairesPage/CommentsButton.js]: Is it worth having an image specifically for this? It's just a circle. A div with `border-radius:100%` would achieve the same and save us a network request

File: src/containers/enhancers/withDeleteQuestionnaire.test.js [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+      {
+        id: 2
+      },
+      {
+        id: 3
+      }
+    ];
+
+    it("should remove the questionnaire from the cache", () => {
+      const readQuery = jest.fn().mockImplementation(({ query }) => {
```
Comment (WickyNilliams) [inline src/containers/enhancers/withDeleteQuestionnaire.test.js]: as before, can be simplified:

```js
jest.fn(() => ({ questionnaires : cachedQuestionnaires }))
```

File: src/containers/enhancers/withDeleteQuestionnaire.test.js [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+
+      handleUpdate(proxy, {
+        data: {
+          deleteQuestionnaire: {
+            id: 1
+          }
+        }
+      });
+
+      expect(readQuery).toHaveBeenCalled();
```
Comment (WickyNilliams) [inline src/containers/enhancers/withDeleteQuestionnaire.test.js]: As before, assert it was called with correct query

File: src/containers/enhancers/withQuestionnaireList.js (hunk 5 above)
Comment (WickyNilliams) [reply src/containers/enhancers/withQuestionnaireList.js]: Seems like YAGNI?

File: src/containers/QuestionnaireCreatePage/index.test.js [hunk 19]
```diff
@@ -57,4 +61,59 @@ describe("containers/QuestionnaireCreatePage", () => {
...
+
+    it("should update the getQuestionnaireList query with new questionnaire.", () => {
+      const newQuestionnaire = { id: 3 };
+      updateQuestionnaireList(proxy, {
+        data: { createQuestionnaire: newQuestionnaire }
+      });
+
+      expect(readQuery).toHaveBeenCalled();
+      expect(writeQuery).toHaveBeenCalledWith(
+        expect.objectContaining({
```
Comment (WickyNilliams) [inline src/containers/QuestionnaireCreatePage/index.test.js]: Probably also worth asserting correct query was supplied.

In my local branch I've been doing it like this:

```js
expect(writeQuery).toHaveBeenCalledWith({
  query: getQuestionnaireList,
  data
});
expect(data).toContain(newQuestionnaire);
```

File: src/containers/enhancers/withDeleteQuestionnaire.test.js [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+
+      expect(writeQuery).toHaveBeenCalledWith(
```
Comment (WickyNilliams) [inline src/containers/enhancers/withDeleteQuestionnaire.test.js]: Again, probably worth asserting writeQuery called with correct query:

```js
expect(writeQuery).toHaveBeenCalledWith({
  query: getQuestionnaireList,
  data
});
expect(data).not.toContain(newQuestionnaire);
```

File: src/containers/QuestionnairesPage/QuestionnairesTable.js (hunk 15 above)
Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesTable.js]: I think this should be done when we have a requirement to display a table elsewhere.

File: src/containers/QuestionnairesPage/QuestionnairesTable.js (hunk 14 above)
Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesTable.js]: @hamishtaplin will look into this to see if we can scale another attribute, e.g. padding.

File: src/containers/QuestionnairesPage/QuestionnairesTable.js [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+TBody.propTypes = {
+  children: PropTypes.arrayOf(PropTypes.element).isRequired
+};
+
+const QuestionnairesTable = ({ questionnaires, onDeleteQuestionnaire }) => {
+  const handleDelete = questionnaireId => {
+    onDeleteQuestionnaire(questionnaireId);
+  };
+
+  if (!questionnaires || questionnaires.length === 0) {
```
[the remark this answers is not in the document]
Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesTable.js]: How is this being handled in other components at the moment? Seems like we'd want to display a spinner or something?

File: src/containers/QuestionnairesPage/QuestionnairesPage.js [hunk 22]
```diff
@@ -1,35 +1,58 @@
...
           <Button id="btn-load-questionnaire" secondary disabled>
-            Load existing questionnaire
+            Upload
           </Button>
-        </ButtonGroup>
-      </CenteredPanel>
-    </Center>
-  </BaseLayout>;
+        </StyledButtonGroup>
+        <CenteredPanel>
```
[the remark this answers is not in the document]
Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesPage.js]: True, it isn't in the design, but the padding is consistent with all of our other pages. Perhaps worth checking with @dsjukes?

File: src/containers/QuestionnairesPage/CommentsButton.js (hunk 16 above)
Comment (samiwel) [reply src/containers/QuestionnairesPage/CommentsButton.js]: Yeah probably not, will remove the asset.

File: src/containers/QuestionnairesPage/QuestionnairesTable.js (hunk 21 above)
[the remark this answers is not in the document]
Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesTable.js]: Done https://github.com/ONSdigital/eq-author/pull/204/commits/0dbacd2f6d7c22e07d1a59d34d760065db8003e0

File: src/containers/QuestionnairesPage/DeleteQuestionnaireButton.js [hunk 23]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import Button from "components/Button";
+
+import DeleteSvg from "./deleteQuestionnaireIcon.svg";
+
+export const StyledButton = styled(Button)`
+  width: 16px;
+  height: 20px;
```
Comment (hamishtaplin) [inline src/containers/QuestionnairesPage/DeleteQuestionnaireButton.js]: This is bigger than in the designs but I think it looks better 👍

File: src/containers/QuestionnairesPage/QuestionnairesTable.js [hunk 24]
```diff
@@ -0,0 +1,177 @@
...
+    border-color: white;
+  }
+`;
+
+const TD = styled.td`
+  text-align: center;
+  line-height: 2;
+  &:nth-child(1),
+  &:nth-child(6) {
+    text-align: left;
```
Comment (hamishtaplin) [inline src/containers/QuestionnairesPage/QuestionnairesTable.js]: Better to handle this with props as removing a column will break it. Also, I think only one column header is center aligned in the designs so maybe a `center` prop?

File: src/containers/QuestionnairesPage/QuestionnairesTable.js [hunk 25]
```diff
@@ -0,0 +1,177 @@
...
+                  </Collapsable>
+                </TD>
+                <TD>
+                  <Collapsable>
+                    {createdAt}
+                  </Collapsable>
+                </TD>
+                <TD>
+                  <Collapsable>
+                    {questionnaire.theme}
```
Comment (hamishtaplin) [inline src/containers/QuestionnairesPage/QuestionnairesTable.js]: This should be in title-case.

Comment (samiwel) [reply src/containers/QuestionnairesPage/QuestionnairesTable.js]: Title case applied with a CSS text-transform.

**Pull request merged**